paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -72,6 +73,9 @@ public Path branchPath(String branchName) {
     }
 
     public void createBranch(String branchName, String tagName) {
+        checkArgument(
+                !branchName.equals(DEFAULT_MAIN_BRANCH),
+                "Branch name 'main' is the default branch and cannot be used.");
         checkArgument(!StringUtils.isBlank(branchName), "Branch name '%s' is blank.", branchName);
         checkArgument(!branchExists(branchName), "Branch name '%s' already exists.", branchName);
         checkArgument(tagManager.tagExists(tagName), "Tag name '%s' not exists.", tagName);
```

Comment:
`tagManager.tagExists` only checks the table path and tag path. I think we should also check the branch path here.

Reply:
The branchExists method already checks the branch path.